utils/plot_correction_files.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+from matplotlib import pyplot as plt
+import glob
+import re
+import math
+import rasterio as rio
+import argparse
+import os
+
+"""
+This script plots the original interferogram, the corresponding correction file, 
+and the resulting corrected interferogram from a PyRate directory with already,
+processed data. directories are given as user arguments to the script, and the
+number of plots is determined by a number range given by user. 
+
+Usage: python3 plot_correction_files.py <IFG_DIR> <CORRECTION_DIR> <CORRECTED_DIR> <SAVE_DIR> <FIRST_IFG> <LAST_IFG>
+
+Command-line arguments:
+           IFG_DIR       - full path to uncorrected interferograms in PyRate.
+    CORRECTION_DIR       - full path to correction files in PyRate.
+     CORRECTED_DIR       - full path to corrected interferograms in PyRate.
+          SAVE_DIR       - full path to directory where images will be saved (needs to exist).
+         FIRST_IFG       - first IFG in range of IFGs to plot (e.g. 1 to start plotting at 1st IFG in directory).
+          LAST_IFG       - last IFG in range of IFGs to plot (e.g. 37 will plot up until the 37th IFG in directory).
+"""
+
+# Arguments
+parser = argparse.ArgumentParser(description="Script to plot orbit correction files with uncorrected and corrected interferogram")
+parser.add_argument("IFG_DIR", type=str, help="full path to uncorrected interferograms in PyRate")
+parser.add_argument("CORRECTION_FILE_DIR", type=str, help="full path to correction files in PyRate")
+parser.add_argument("CORRECTED_IFG_DIR", type=str, help="full path to corrected interferograms in PyRate")
+parser.add_argument("SAVE_DIR", type=str, help="full path to directory where images will be saved")
+parser.add_argument("FIRST_IFG", type=int, help="first IFG in range of IFGs to plot (e.g. 1 to start plotting at first IFG in directory)")
+parser.add_argument("LAST_IFG", type=int, help="last IFG in range of IFGs to plot (e.g. 37 will plot up until the 37th IFG in directory)")
+args = parser.parse_args()
+
+
+
+# Directories and variable from user agruments
+ifg_dir = os.path.abspath(args.IFG_DIR)
+corr_dir = os.path.abspath(args.CORRECTION_FILE_DIR)
+tempml_dir = os.path.abspath(args.CORRECTED_IFG_DIR)
+save_dir = os.path.abspath(args.SAVE_DIR)
+
+first_ifg_num = args.FIRST_IFG - 1
+last_ifg_num = args.LAST_IFG - 1
+
+
+# Create Lists
+ifg_list = []
+for file in glob.glob(f'{ifg_dir}/*ifg.tif'):
+    ifg_list.append(file)
+
+corr_list = []
+for file in glob.glob(f'{corr_dir}/*.npy'):
+    corr_list.append(file)
+
+tempml_list = []
+for file in glob.glob(f'{tempml_dir}/*ifg.tif'):
+    tempml_list.append(file)
+
+# Sort
+ifg_list.sort()
+corr_list.sort()
```

Comment:
Are the contents of ifg_list, corr_list and tempml_list guaranteed to be from an identical set of date pairs? (i.e. do the corrections ever fail and result in no corrections being saved for one or more of the ifgs?)

If not, the sort-all-lists approach here may result in mismatching ifgs to corrections in the plots.

Reply:
Yeah good point, I look into this before merging.

Reply:
This change has been made by comparing the date-pairs of the three datasets per figure.